src/python/WMCore/Storage/TrivialFileCatalog.py

```diff
@@ -184,9 +185,44 @@ def tfcProtocol(contactString):
     """
     args = urlsplit(contactString)[3]
     value = args.replace("protocol=", '')
+    if not useTFC:
+        value = value.split('&')[0]
     return value
 
+#construct a catalog from storage attribute  
+def getCatalog(storageAttr):
+    site = storageAttr['site']
+    subSite = storageAttr['subSite']
+    storageSite = storageAttr['storageSite']
+    #get site config
+    siteconfig_path = os.getenv('SITECONFIG_PATH',None)
+    if not siteconfig_path:
+        raise RuntimeError('SITECONFIG_PATH is not defined')
+    subPath = ''
+    #not a cross site, use local path given in SITECONFIG_PATH
```

Comment:
Can you please clarify what a cross site is? Is it a site that shares the same RSE with other sites (e.g.: T2_CH_CERN x T2_CH_CERN_P5 x T2_CH_CERN_HLT, all sharing the T2_CH_CERN RSE)

Reply:
I do not have a deep understanding of how all sites are connected. The idea here is that jobs can be executed at a site I called it **local** and the outputs (stage out?) can be stored at other sites which I called **cross site** here. As I explained above, in this **cross site** case the ```site``` attribute in a ```<stage-out>``` subnode is explicitly defined. If it is not **cross site** case, the storage site is **local**. Therefore it is the same as local site (```if site == storageSite```)